nwss_wastewater/delphi_nwss/run.py

```diff
@@ -126,38 +78,52 @@ def run_module(params):
     export_dir = params["common"]["export_dir"]
     socrata_token = params["indicator"]["socrata_token"]
     if "archive" in params:
-        daily_arch_diff = S3ArchiveDiffer(
+        arch_diff = S3ArchiveDiffer(
             params["archive"]["cache_dir"],
             export_dir,
             params["archive"]["bucket_name"],
-            "nchs_mortality",
+            "nwss_wastewater",
             params["archive"]["aws_credentials"],
         )
-        daily_arch_diff.update_cache()
+        arch_diff.update_cache()
 
     run_stats = []
     ## build the base version of the signal at the most detailed geo level you can get.
     ## compute stuff here or farm out to another function or file
-    df_pull = pull_nwss_data(socrata_token)
-    ## aggregate
-    for sensor in SIGNALS:
-        df = df_pull.copy()
-        # add weighed column
-        df = generate_weights(df, sensor)
-
-        for geo in GEOS:
-            logger.info("Generating signal and exporting to CSV", metric=sensor)
-            if geo == "nation":
-                agg_df = weighted_nation_sum(df, sensor)
-            else:
-                agg_df = weighted_state_sum(df, geo, sensor)
-            # add se, sample_size, and na codes
-            agg_df = add_needed_columns(agg_df)
-            # actual export
-            dates = create_export_csv(
-                agg_df, geo_res=geo, export_dir=export_dir, sensor=sensor
-            )
-            if len(dates) > 0:
-                run_stats.append((max(dates), len(dates)))
+    df_pull = pull_nwss_data(socrata_token, logger)
+    geomapper = GeoMapper()
+    # iterate over the providers and the normalizations that they specifically provide
+    for provider, normalization in zip(PROVIDER_NORMS["provider"], PROVIDER_NORMS["normalization"]):
+        # copy by only taking the relevant subsection
+        df_prov_norm = df_pull[(df_pull.provider == provider) & (df_pull.normalization == normalization)]
+        df_prov_norm = df_prov_norm.drop(["provider", "normalization"], axis=1)
+        for sensor in [*SIGNALS, *METRIC_SIGNALS]:
+            full_sensor_name = sensor + "_" + provider + "_" + normalization
+            df_prov_norm = df_prov_norm.rename(columns={sensor: full_sensor_name})
+            for geo in GEOS:
+                logger.info("Generating signal and exporting to CSV", metric=full_sensor_name)
+                if geo == "nation":
+                    df_prov_norm["nation"] = "us"
+                agg_df = geomapper.aggregate_by_weighted_sum(
```

Comment:
question: looks like we're aggregating for all geos (state and nation). Is the base geo type not reportable?

Reply:
The base geo is `(state, wwtp_id)` (wastewater treatment plant id), which doesn't really match any other source at all. I was planning on waiting to add that, possibly outside of covidcast-indicators. Do you think just adding it to covidcast-indicators is a good idea?

Reply:
Ah, wastewater treatment plant id seems not worth adding; I don't think it would be particularly useful to users since it's a nonstandard geo and isn't clear how it corresponds to counties, MSAs, etc.

If we were to add wwtp_id, I'd think we'd put it in covidcast, too. We haven't split an indicator across endpoints by geo.